### ISSUE TITLE
Cannot build documentation of a package that depends on ospsuite-r

### ISSUE THREAD

**Issue by PavelBal** [Open-Systems-Pharmacology/OSPSuite-R#507]
For example if I try to create documentation for the Reporting Engine:

```
==> devtools::document(roclets = c('rd', 'collate', 'namespace'))

i Updating ospsuite.reportingengine documentation
i Loading ospsuite.reportingengine
Error: object 'getCurrentConvertedObject' not found
Error: Dependency package(s) 'ospsuite' not available.
Backtrace:
    x
 1. +-base::suppressPackageStartupMessages(...)
 2. | \-base::withCallingHandlers(expr, packageStartupMessage = function(c) tryInvokeRestart("muffleMessage"))
 3. \-devtools::document(roclets = c("rd", "collate", "namespace"))
 4.   \-roxygen2::roxygenise(pkg$path, roclets)
 5.     \-roxygen2:::load_code(base_path)
 6.       \-pkgload::load_all(path, helpers = FALSE, attach_testthat = FALSE)
 7.         \-pkgload:::load_depends(path)
 8.           \-pkgload:::abort_for_missing_packages(res, depends$package)
Warning message:
In (function (dep_name, dep_ver = "*")  :
  Dependency package 'ospsuite' not available.
Execution halted

Exited with status 1.
```

The problem lies in calling a static method during onLoad():

https://github.com/Open-Systems-Pharmacology/OSPSuite-R/blob/a2c8352731e39d218206e4d72178eede3c73cc9f/R/init-package.R#L22

I was trying to understand what is happening and it seems that rClr is not directly loaded when devtools or roxygen tries to generate the documentation.

A solution is to put

`library(rClr)` into the .onLoad-function of ospsuite-r:

```
.onLoad <- function(...) {

  # Only for x64 bits packages. This is required to avoid error when package is being checked on CI for x86
  is64 <- (.Machine$sizeof.pointer == 8)
  if (!is64) {
    return()
  }

  library(rClr)
  initPackage()
}
```

though I think I have read it multiple times that using `library()` directly in the package is  veeeery bad style (no idea why). Any thoughts?

**Comment by msevestre** [comment]
@PavelBal But we can create the documentation for the reporting engine when we build the package on appveyor....
Somehow this works

**Comment by msevestre** [comment]
oh that's new...

**Comment by msevestre** [comment]
Isn't it possiblet to load rclrc Explicitely in reportingengine instead as dependency?

**Comment by PavelBal** [comment]
> Isn't it possiblet to load rclrc Explicitely in reportingengine instead as dependency?

Maybe, but then it must be done in every package that depends on ospsuite-r. I was debugging my own package the whole day until I had an idea to try to build documentation for the reporting engine...

**Comment by PavelBal** [comment]
So the documentation for reportingengine cannot be buit, right?

**Comment by msevestre** [comment]
No it's building no problem as far as I can tell from the build...

**Comment by PavelBal** [comment]
But the documentation is not updated, check from:

https://ci.appveyor.com/project/open-systems-pharmacology-ci/OSPSuite-ReportingEngine/branch/develop#L5670

At the end, documentation for new methods `addXParametersForPkParametersPlot` and `addYParametersForPkParametersPlot` is not created.

**Comment by msevestre** [comment]
Maybe this needs to be in the NAMESPACE?
https://ci.appveyor.com/project/open-systems-pharmacology-ci/OSPSuite-ReportingEngine/branch/develop#L5693?

**Comment by PavelBal** [comment]
NAMESPACE is usually generated by Roxygen, but Roxygen fails directly. The only way how I could solve it is explicitly calling  `library(rClr)` before `.onLoad()`.

Documentation always fails is ospsuite-r is in the depends-list, even with a completely empty project (create a new empty project  -> put ospsuite-r into depends -> try to generate documentation -> fail).

**Comment by PavelBal** [comment]
This has something to do with rClr or the methods that are called. Calling 

`  rClr::clrCallStatic("OSPSuite.R.Api", "InitializeOnce", apiConfig$ref)`

in `initPackage` works, calling  
` rClr::clrCallStatic("OSPSuite.R.Api", paste0("GetDimensionTask"))` fails. Actullay everything that would return a clr object fails at this stage.

**Comment by msevestre** [comment]
What happens if we add rclrc to the DESCRIPTION of the package explicitly?

**Comment by msevestre** [comment]
like this
https://github.com/Open-Systems-Pharmacology/OSPSuite-R/blob/develop/DESCRIPTION#L20

**Comment by PavelBal** [comment]
Nope.

**Comment by PavelBal** [comment]
https://github.com/r-lib/roxygen2/issues/1224

**Comment by PavelBal** [comment]
From https://github.com/r-lib/roxygen2/issues/1224#issuecomment-823233282

We should add

`Roxygen: { library(PackageB) ; list() }`

to the `DESCRIPTION` of packages that depend on ospsuite-r (ReportingEngine etc).

**Comment by msevestre** [comment]
what about this thread here
https://github.com/r-lib/roxygen2/issues/1224#issuecomment-823247129
?

**Comment by msevestre** [comment]
https://github.com/r-lib/pkgload/issues/156

and that specifically. We need to import differently then?
=

**Comment by PavelBal** [comment]
> what about this thread here

This is the thread I opened :D I will ask about importing.

**Comment by PavelBal** [comment]
That is a very weird case. As discussed in the issue, using 

`PackageA::helloA()` instead of simply `helloA()` in PackageB solves the issue. However, we already always explicitly call `rClr::methodName()`!

**Comment by PavelBal** [comment]
OK found it. At some point, rClr envokes a method in R from .NET, where the method is without `rClr::`. Is fixed in the last version if rClr (0.9.1), so the workaround would not be necessary.

**Comment by PavelBal** [comment]
This will not be an issue with rClr >= 0.9.1.
